common/lib/xmodule/xmodule/modulestore/tests/factories.py

```diff
@@ -369,13 +369,14 @@ def _create(cls, target_class, **kwargs):  # lint-amnesty, pylint: disable=argum
         # and doesn't need to be passed into the object constructor
         kwargs.pop('descriptive_tag')
 
+        parent = kwargs.pop('parent', None)
+        parent_location = parent.location if parent else parent_location
+
         assert isinstance(location, UsageKey)
         assert location != parent_location
 
         store = kwargs.pop('modulestore')
-
-        # This code was based off that in cms/djangoapps/contentstore/views.py
-        parent = kwargs.pop('parent', None) or store.get_item(parent_location)
+        parent = store.get_item(parent_location)  # always re-get the parent from the store
```

Comment:
This lets us more reliably just pass `parent=` in instead of a location - we'll always get a fresh lookup from the store.